addons/web/models/ir_http.py

```diff
@@ -190,3 +194,6 @@ def _placeholder(self, image=False):
             image = 'web/static/img/placeholder.png'
         with file_open(image, 'rb', filter_ext=('.png', '.jpg')) as fd:
             return fd.read()
+
+    def _check_action_exists(self, action_id):
+        return self.env['ir.actions.actions'].sudo().browse([action_id]).exists()
```

Comment:
The entire thing is unnecessary, the caller can just check if `user.action_id.exists()` (or `home_action.exists()`).